### ISSUE TITLE
why scalar Gaussian only?

### ISSUE THREAD

**Issue by moquan** [mattjj/pyhsmm-factorial#5]
Hi Matt
This might be a long shot, but may I ask why was only scalar Gaussian allowed in your factorial example? Is it because high dimensional data is computationally impossible? Or was it because it was a trivial further extension? I am trying to apply HDP-FHSMM on multi-dimensional data, and if you could provide some tips that would be much appreciated. Thank you very much!

Sincerely
Moquan

**Comment by mattjj** [comment]
I implemented it in 1D just because that's what I needed at the time, and so I special-cased a few computations  (like [resampling component emissions](https://github.com/mattjj/pyhsmm-factorial/blob/master/internals/eigen_sample_component_emissions.cpp) and maybe some other places where I left [comments](https://github.com/mattjj/pyhsmm-factorial/blob/master/internals/states.py#L165)).

However, having higher-dimensional emissions shouldn't affect the inference complexity much (the corresponding likelihood computations and resampling steps probably wouldn't be the bottleneck until the dimension is at least 1000 or so, to guess a number).

To handle the higher-dimensional case, one way to start would be to look for where I wrote comments and/or asserts that include the word 'scalar'. I'm not sure if those were exhaustive, though, and I haven't looked at this code in a very long time (I see I wrote it when I was still [using weave](https://github.com/mattjj/pyhsmm-factorial/blob/master/internals/states.py#L126) instead of cython).

It's unlikely that I'll be returning to this code in the near future, but I'll answer any questions you have as best I can.
